services-core/src/economic_viability.rs

```diff
@@ -10,6 +10,59 @@ use std::{num::NonZeroU128, sync::Arc};
 /// much gas is used in the reversion of the previous solution.
 const GAS_PER_TRADE: f64 = 120_000.0;
 
+#[derive(Debug)]
+pub enum EconomicViabilityStrategy {
+    Dynamic,
+    Static,
+}
+
+impl std::str::FromStr for EconomicViabilityStrategy {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Dynamic" => Ok(Self::Dynamic),
+            "Static" => Ok(Self::Static),
+            _ => Err("invalid value"),
+        }
+    }
+}
+
+impl EconomicViabilityStrategy {
+    pub fn variants() -> &'static [&'static str] {
+        &["Dynamic", "Static"]
+    }
```

Comment:
Its a pity there is no macro for generating this without pulling in `clap`.

Reply:
nit: Maybe this method and `from_str` could depend on the same Map so that `variants` is equal to `keys` and `from_str` is just a lookup.

Reply:
I think we can use #1413 to do this.